motor_control/throttle/base_throttle.h

```diff
@@ -6,11 +6,12 @@ class Throttle
 public:
     Throttle(const int &pin);
 
-    virtual void compute_motor_value() = 0; // this needs to be implemented differently for variable/button throttle
+    virtual int compute_motor_value() = 0; // this needs to be implemented differently for variable/button throttle
 
     void read_sensor_value();
     int get_raw_value() const; // should only be used to debug
     int get_motor_value() const;
+    int get_pin() const;
```

Comment:
Why would we need to expose this? Dealing with pin IO should be contained to within the class.

Reply:
I needed the pin number for a method on the derived classes but it's a private member - also, should I make m_motor_value a protected value as opposed to a private one to reduce the amount of one line getter and setter methods?

Reply:
I like protected variables instead of having accessor and modifier methods when dealing with inherited classes. Happy to let @abdullahelqaq give the final judgement

Reply:
I see below that you wanted access to the pin to do a `digitalRead()` instead of an `analogRead()`. In this case, we should change the base class to give the option (by setting a flag) to use `digitalRead()` instead. You'll remember from our previous discussion that we want to isolate Arduino-specific code as much as we can, and it still holds true here. I can put up a PR with the changes you'll need to avoid having to reference the pin directly.

Reply:
I just put the [PR](https://github.com/Bruin-SuperMileage/EV/pull/18) up. As soon as it lands, please pull and marge the changes into your branch so that you can use it. Feel free to reach out to myself or @ashvinnagarajan with help on how to do that if you need it!